### DISCUSSION TITLE
Baffling error: TypeError: Cannot read properties of undefined (reading 'isDynamic')

### DISCUSSION THREAD

**Discussion by amaschas** [vercel/next.js#40373] [Help] [answered]
Has anyone run across this error when using `getServerSideProps`?

```
  TypeError: Cannot read properties of undefined (reading 'isDynamic')
      at Object.sendRenderResult (/<path>/node_modules/next/server/send-payload/index.ts:65:26)
      at DevServer.sendRenderResult (/<path>/node_modules/next/server/next-server.ts:657:7)
      at DevServer.pipe (/<path>/node_modules/next/server/base-server.ts:827:19)
```

I'm having a hard time even nailing down what causes it, but it appears to be related to what props get returned from `getServerSideProps`. Our currently looks like this:

```
export const getServerSideProps = async (context: GetServerSidePropsContext) => {
    let initialAppParams: InitialAppParams = {
        apiArticle: null,
        path: ""
    };
    initialAppParams = {
        ...initialAppParams,
        ...await setInitialAppParams(context)
    };

    return {
        props: {
            ...initialAppParams,
        }
    };
};
```

There we make an API request in `setInitialAppParams` and grab the value of `context.resolvedUrl`, but beyond that we're not doing anything particularly special. We are using a custom server, but the code we use in that case is completely boilerplate. I'm sort of at a loss for what could be causing this error, and the actual text of the error is not particularly illuminating. I've searched everywhere for a reference to errors involving `isDynamic` and found nothing. Any clues for where to look would be appreciated.

**Comment by amaschas** [comment]
OK, so I found the answer to this and I believe this may be a bug. The problem in this case was that a component in the tree was importing a JSON file, and something about our configuration is incorrect somehow. But the bigger problem is that Next doesn't report the actual error, but rather throws this odd error about `isDynamic`. I'll create a bug ticket for this.

**Comment by jacknevitt** [comment]
I am also getting this error. Seems to appear when I throw an error inside `getServerSideProps`. Not sure why my own error message is nowhere to be seen. Nextjs version 12.2.4.

**Reply by amaschas** [reply]
Yeah I think the problem here is not so much the `isDynamic` error, but rather the fact that it suppresses the actual error. Still haven't found a solution for this.

**Reply by jacknevitt** [reply]
Does the isDynamic error still appear when you build your app and run it?

**Reply by amaschas** [reply]
It comes and goes. Basically, if there is an error in `getServerSideProps`, the `isDynamic` error gets throw instead of whatever error is actually occurring.

**Reply by jacknevitt** [reply]
For me it was also when errors were thrown during render, gSSP was successful, but the server could not render the page (due to me trying to render an object as a react child, for example). Definitely still an issue with the original error not being shown correctly though

**Reply by jacknevitt** [reply]
We're using a custom router file on this project. One thought I had was that this error is shown when trying to display the error page.

**Reply by amaschas** [reply]
Yeah, I see the same issue where it tries to render the error page after failed on the initially requested page, then throws the `isDynamic` error. I'm also pretty sure this only started happening within the last few versions of NextJS.

**Comment by amaschas** [comment]
Just a quick update, I reported a bug here but I'm struggling to isolate the error in an example repo. If anyone else who is running into this problem has had better luck please respond to the ticket: https://github.com/vercel/next.js/issues/41540

**Comment by amaschas** [comment, marked as answer]
For anyone else running into this issue: I narrow it down to the use of the `dd-trace` plugin. Once I removed the `import "dd-trace/init";` line, the isDynamic errors disappeared and the actual error was reported correctly. I'll open an issue on the dd-trace end and link it here.

Correction: we had a severely out of date version of the dd-trace plugin (2.12.2). Using the latest version 3.5.0 fixes the problem.
